services/api/src/utils/schema.js

```diff
@@ -68,21 +69,109 @@ function createSchema(attributes = {}, options = {}) {
     });
   });
 
-  schema.methods.assign = function assign(fields) {
+  schema.method('assign', function assign(fields) {
     for (let [key, value] of Object.entries(fields)) {
       if (!value && isReferenceField(this.schema.obj[key])) {
         value = undefined;
       }
       this[key] = value;
     }
-  };
+  });
+
+  // Soft delete
+
+  schema.pre(/^find|count|exists/, function (next) {
+    const filter = this.getFilter();
+    if (filter.deletedAt === undefined) {
+      if ('deletedAt' in filter) {
+        delete filter.deletedAt;
+      } else {
+        filter.deletedAt = { $exists: false };
+      }
+    }
+    return next();
+  });
 
-  schema.methods.delete = function () {
+  schema.method('delete', function () {
```

Comment:
whats the diff ? just style

Reply:
yeah just trying to align our style here (instance and static) ... could be the other way as well if you like it better